components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/webnotifications/GeckoWebNotificationDelegate.kt

```diff
@@ -0,0 +1,29 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.webnotifications
+
+import mozilla.components.concept.engine.webnotifications.WebNotification
+import mozilla.components.concept.engine.webnotifications.WebNotificationDelegate
+import org.mozilla.geckoview.WebNotification as GeckoViewWebNotification
+import org.mozilla.geckoview.WebNotificationDelegate as GeckoViewWebNotificationDelegate
+
+internal class GeckoWebNotificationDelegate(
+    private val webNotificationDelegate: WebNotificationDelegate
+) : GeckoViewWebNotificationDelegate {
+    override fun onShowNotification(webNotification: GeckoViewWebNotification) {
+        webNotificationDelegate.onShowNotification(webNotification.toWebNotification())
+    }
+
+    override fun onCloseNotification(webNotification: GeckoViewWebNotification) {
+        webNotificationDelegate.onCloseNotification(webNotification.toWebNotification())
+    }
+
+    private fun GeckoViewWebNotification.toWebNotification(): WebNotification {
+        this.apply {
```

Comment:
`this.apply` is unneeded since its an extension method

Reply:
Good catch, forgot that I left that there.  Thanks,